src/explicitode.jl

```diff
@@ -112,48 +112,80 @@ Returns a maximum time-step for a the Taylor expansion `x`
 using a prescribed absolute tolerance `epsilon` and the last two
 Taylor coefficients of (each component of) `x`.
 
-Note that `x` is of type `Taylor1{T}` or `Vector{Taylor1{T}}`, including
-also the cases `Taylor1{TaylorN{T}}` and `Vector{Taylor1{TaylorN{T}}}`.
+Note that `x` is of type `Taylor1{U}` or `Vector{Taylor1{U}}`, including
+also the cases `Taylor1{TaylorN{U}}` and `Vector{Taylor1{TaylorN{U}}}`.
 
+Depending of `eltype(x)`, i.e., `U<:Number`, it may be necessary to overload
+`stepsize`, specializing it on the type `U`, to avoid type instabilities.
 """
 function stepsize(x::Taylor1{U}, epsilon::T) where {T<:Real, U<:Number}
+    R = promote_type(typeof(norm(constant_term(x), Inf)), T)
     ord = x.order
-    h = convert(T, Inf)
+    h = convert(R, Inf)
+    z = zero(R)
     for k in (ord-1, ord)
         @inbounds aux = norm( x[k], Inf)
-        aux == zero(T) && continue
-        aux = epsilon / aux
-        kinv = one(T)/k
-        aux = aux^kinv
-        h = min(h, aux)
+        aux == z && continue
+        aux1 = _stepsize(aux, epsilon, k)
+        h = min(h, aux1)
     end
-    isfinite(h) && return h
-
-    # If isinf(h) == true, we use the maximum (finite)
-    # step-size obtained from all coefficients as above.
-    # Note that the time step is independent from `epsilon`
-    hh = zero(T)
-    for k in 1:ord-1
-        @inbounds aux = norm( x[k], Inf)
-        aux == zero(T) && continue
-        aux = 1 / aux
-        kinv = one(T)/k
-        aux = aux^kinv
-        hh = max(hh, aux)
-    end
-    hh == zero(T) && return h
-
-    return hh
+    return h::R
 end
 
 function stepsize(q::AbstractArray{Taylor1{U},1}, epsilon::T) where
         {T<:Real, U<:Number}
-    h = convert(T, Inf)
+    R = promote_type(typeof(norm(constant_term(q[1]), Inf)), T)
+    h = convert(R, Inf)
     for i in eachindex(q)
         @inbounds hi = stepsize( q[i], epsilon )
         h = min( h, hi )
     end
-    return h
+
+    # If `isinf(h)==true`, we use the maximum (finite)
+    # step-size obtained from all coefficients as above.
+    # Note that the time step is independent from `epsilon.
```

Comment:
`` `epsilon `` -> `` `epsilon` ``

Reply:
Good catch, I'll push a new commit correcting this.